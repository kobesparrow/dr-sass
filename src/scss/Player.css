.nameInput {
  width: 400px;
  border: 2px solid #082B33;
  border-radius: 25px;
  background: #FFF;
  margin-left: 25%;
  margin-bottom: 300px;
  margin-top: 50px;
}

.nameDisplay {
  width: 400px;
  border: 2px solid #082B33;
  border-radius: 25px;
  background: #FFF;
  margin-left: 25%;
  margin-bottom: 300px;
  margin-top: 50px;
}

.intro-lorax {
  height: 150px;
}

.name-submit-btn {
  margin-bottom: 30px;
}