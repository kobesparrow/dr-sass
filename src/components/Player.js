import React, { Component } from 'react';
import '../scss/Player.scss';
import lorax from '../images/lorax-book.jpg'

class Player extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    let introText
    if (this.props.nameSubmitted) {
      introText = 
        <article className="nameDisplay">
          <p>Welcome { this.props.name }</p>
          <p>Score: { this.props.playerScore }</p>
          <p>{ this.props.answerStatus }</p>
        </article>
    } else {
      introText = 
      <article className="nameInput">
        <p>What is your name?</p>
        <input type="text" onChange={ this.props.setName } name="question" id="name" placeholder="Lorax"></input>
        <input onClick={ () => this.props.submitName() } type="submit" name="submit" className="name-submit-btn"></input>
        <img src={ lorax } className="intro-lorax" />
      </article>
    }
    return (
      <div>{ introText }</div>
    );
  }
}

export default Player;