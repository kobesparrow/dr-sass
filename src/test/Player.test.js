import React from 'react';
import Player from '../components/Player';
import { shallow } from 'enzyme';

let mockData = {
  nameSubmitted: false,
  name: "Mark",
  answerStatus: 'You win',
  playerScore: 0
}
let mockSubmitName = jest.fn();
let mockSetName = jest.fn();

describe('Player', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Player
        submitName={mockSubmitName}
        setName={mockSetName}
        nameSubmitted={mockData.nameSubmitted}
        name={mockData.name}
        answerStatus={mockData.answerStatus}
        playerScore={mockData.playerScore}
      />
    )
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should meet the first condition', () => {
    wrapper = shallow(
      <Player nameSubmitted={true} />
    )
    expect(wrapper).toMatchSnapshot();
  });
  it('should invoke submitName on click', () => {
    wrapper.find('.name-submit-btn').simulate('click');
    expect(mockSubmitName).toHaveBeenCalled();
  })
});