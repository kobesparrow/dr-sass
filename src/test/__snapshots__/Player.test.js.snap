// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should match the snapshot 1`] = `
<div>
  <article
    className="nameInput"
  >
    <p>
      What is your name?
    </p>
    <div
      className="submit-name-block"
    >
      <form>
        <input
          id="name"
          name="question"
          onChange={[MockFunction]}
          placeholder="Lorax"
          type="text"
        />
        <input
          className="name-submit-btn"
          name="submit"
          onClick={[Function]}
          type="submit"
        />
      </form>
      <img
        className="intro-lorax"
        src="lorax-book.jpg"
      />
    </div>
  </article>
</div>
`;

exports[`Player should meet the first condition 1`] = `
<div>
  <article
    className="nameDisplay"
  >
    <p>
      Welcome 
    </p>
    <p>
      Score: 
    </p>
    <p />
  </article>
</div>
`;
